frontend/src/old-pages/Configure/SlurmSettings.tsx

```diff
@@ -0,0 +1,174 @@
+    i18next.t('wizard.headNode.slurmSettings.validation.passwordCannotBeEmpty'),
+  ]
+
+  if (errorMask.every(e => e) || errorMask.every(e => !e)) {
+    return true
+  } else {
+    slurmAccountingSetErrors(errorMask, errorPaths, errorValues)
+    return false
+  }
+}
+
+function slurmAccountingValidateField(field: string | undefined): boolean {
+  return field ? true : false
+}
+
+function slurmAccountingSetErrors(
+  errorMask: Array<boolean>,
+  errorPaths: Array<Array<string>>,
+  errorValues: Array<string>,
+) {
+  errorMask.forEach((err, idx) =>
+    err
+      ? clearState(errorPaths[idx])
+      : setState(errorPaths[idx], errorValues[idx]),
+  )
+}
+
+function DatabaseField() {
+  const {t} = useTranslation()
+  let uri = useState(uriPath) || ''
+  let uriError = useState(uriErrorPath) || ''
+
+  return (
+    <FormField
+      label={t('wizard.headNode.slurmSettings.database.label')}
+      errorText={uriError}
+    >
+      <Input
+        onChange={({detail}) => {
+          setState(uriPath, detail.value)
+        }}
+        value={uri}
+        placeholder={t('wizard.headNode.slurmSettings.database.placeholder')}
+      />
+    </FormField>
+  )
+}
+
+function UsernameField() {
+  const {t} = useTranslation()
+  let username = useState(usernamePath) || ''
+  let usernameError = useState(usernameErrorPath) || ''
+
+  return (
+    <FormField
+      label={t('wizard.headNode.slurmSettings.username.label')}
+      errorText={usernameError}
+    >
+      <Input
+        onChange={({detail}) => {
+          setState(usernamePath, detail.value)
+        }}
+        value={username}
+        type="text"
+      />
+    </FormField>
+  )
+}
+
+function PasswordField() {
+  const {t} = useTranslation()
+  let password = useState(passwordPath) || ''
+  let passwordError = useState(passwordErrorPath) || ''
+
+  return (
+    <FormField
+      label={t('wizard.headNode.slurmSettings.password.label')}
+      errorText={passwordError}
+    >
+      <Input
+        onChange={({detail}) => {
+          setState(passwordPath, detail.value)
+        }}
+        value={password}
+        type="text"
+      />
+    </FormField>
+  )
+}
+
+function SlurmSettings() {
+  const {t} = useTranslation()
+
+  return (
+    <Container
+      header={
+        <Header variant="h2">
+          {t('wizard.headNode.slurmSettings.container.title')}{' '}
+          <span
+            style={{fontWeight: '400', fontStyle: 'italic', fontSize: '14px'}}
+          >
+            - {t('wizard.headNode.slurmSettings.container.optional')}
+          </span>
+        </Header>
+      }
+    >
+      <ColumnLayout columns={2}>
+        <DatabaseField />
+      </ColumnLayout>
+      <ColumnLayout columns={2}>
+        <UsernameField />
+        <PasswordField />
+      </ColumnLayout>
+    </Container>
+  )
+}
+
+export {
+  SlurmSettings,
+  slurmAccountingValidateAndSetErrors,
+  slurmAccountingValidateField,
+  slurmAccountingSetErrors,
```

Comment:
suggestion(non-blocking): no big deal, in the future you could export these functions from a different file, otherwise you could end up with weird un-mocked imports in your tests (e.g. if `SlurmSettings` had side-effectful imports, you'd be forced to mock them in the validation functions' tests)